xbmc/pvr/PVRGUIActions.cpp

```diff
@@ -647,22 +661,19 @@ namespace PVR
     {
       if (newTimer->GetTimerType() == timer->GetTimerType())
       {
-        return CServiceBroker::GetPVRManager().Timers()->UpdateTimer(newTimer);
+        if (CServiceBroker::GetPVRManager().Timers()->UpdateTimer(newTimer))
+          return true;
+
+        HELPERS::ShowOKDialogText(CVariant{257}, CVariant{19263}); // "Error", "Could not update the timer. Check the log for more information about this message."
+        return false;
       }
       else
       {
         // timer type changed. delete the original timer, then create the new timer. this order is
         // important. for instance, the new timer might be a rule which schedules the original timer.
         // deleting the original timer after creating the rule would do literally this and we would
         // end up with one timer missing wrt to the rule defined by the new timer.
-        if (CServiceBroker::GetPVRManager().Timers()->DeleteTimer(timer, timer->IsRecording(), false))
-        {
-          if (AddTimer(newTimer))
-            return true;
-
-          // rollback.
-          return AddTimer(timer);
-        }
+        return DeleteTimer(timer, timer->IsRecording(), false);
```

Comment:
Looks like the long comment above this block will need updating

Reply:
Ooops. The comment is correct, but I deleted some stuff that has to be kept.